packages/website/blog/2025-05-26-project-service.mdx

```diff
@@ -0,0 +1,226 @@
+
+When we first started working with the new project service API, it outperformed equivalent `parserOptions.project` setups by ~10-15%.
+Since then, we have observed regressions in performance that have brought it down in some cases to be slightly slower.
+
+We believe the new Project Service API should be faster than the traditional API, and are treating the lack of significant improvement as a bug.
+See [⚡ Performance: parserOptions.projectService sometimes no longer outperforms parserOptions.project](https://github.com/typescript-eslint/typescript-eslint/issues/9571) for more information.
+
+## Next Steps
+
+The new Project Service API is available in typescript-eslint v8.0.0 and later.
+We've been using it in our monorepo for over a year and have been thrilled to see many community repositories adopt it as well.
+
+### Giving Feedback
+
+We'd love to hear from you on how this option works for you.
+Does it live up to what we've promised, and/or does it have bugs we haven't fixed yet?
+Please do send us GitHub issues for any bugs you encounter or suggestions for how to improve the API.
+
+The [typescript-eslint Discord](https://discord.gg/FSxKq8Tdyg) is a great place to ask questions and engage with us more casually.
+For support in onboarding, feel free to ask in its `#help` channel.
+We'd be happy to help you try out `parserOptions.projectService` and learn more about how you use typescript-eslint.
+
+### Long Term Vision
+
+The new Project Service API is a great step towards making typed linting easier and more straightforward to configure.
+Our priority for the next year will be to improve the new Project Service API so that it works in all places the traditional Program API does.
+We won't remove the traditional project program behavior unless and until the new Project Service API is able to fully replace it.
+
+As of typescript-eslint@8.33.0, we've also extracted most of the Project Service code into a standalone [`@typescript-eslint/project-service`](/packages/project-service) package.
+It has no dependencies on ESLint and is designed to be usable for any linter to enable TypeScript's Project Service API for typed linting.
+See [Packages > Project Service](/packages/project-service) for more details.
+
+We're also looking forward to investigating support for [TypeScript's 10x faster Go port](https://github.com/typescript-eslint/typescript-eslint/issues/10940).
+Abstracting configuration details into the Project Service API means it will be much easier for typescript-eslint to support typed linting using "tsgo" without any additional user configuration.
+
+So, please, try out the new Project Service API.
+We're excited to hear how it works for you and what we can do to improve it. 💜
+
+[^1]: https://github.com/babel/babel/pull/16192#issue-2054613116
+
+[^2]: https://github.com/t3-oss/create-t3-app/pull/1936/#discussion_r1667389041
```

Comment:
in case you care for any more testimonials, I just benchmarked `project` vs `projectService` on the SvelteKit repo and `projectService` came out ahead there, so I sent a PR to switch over: https://github.com/sveltejs/kit/pull/13839

Reply:
Ahhhhh this is fantastic, thank you so much @benmccann! Tbh I was pretty upset that we didn't have many good references of public speed-ups. This is _really, really_ good. Thank you!!